create-editor.js

```diff
@@ -53,7 +53,32 @@ require(["vs/editor/editor.main"], function () {
     editor = monaco.editor.create(document.getElementById("container"), {
         value: jsCode,
         language: "typescript",
-        theme: "normal"
+        theme: "normal",
+        formatOnType: true,
+        formatOnPaste: true,
+    });
+
+    monaco.languages.typescript.typescriptDefaults.setCompilerOptions({
+        /* Disable allowNonTsExtensions to remove keywords from autocomplete
+         * This introduces an error in the console. Unknown if this will have a negative impact when compiling.
+         */
+        allowNonTsExtensions: false,      
+        noLib: true
+    });
+
+    monaco.languages.registerCompletionItemProvider("typescript", {
+        provideCompletionItems: function() {
+            var autocomplete = [];
+            for (var i = 0; i < blockDict.length; i++) {
+                autocomplete[i] = {
+                    label: blockDict[i]["blockName"],
+                    kind: blockDict[i]["kind"],
+                    documentation: blockDict[i]["documentation"],
+                    insertText: blockDict[i]["code"]
```

Comment:
I've changed my mind about using this syntax (sorry!). Please replace with: 

```javascript
label: blockDict[i].blockName,
```

etc.